### PR TITLE
Feature/eodhp 280 nested catalog support in stac api

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
## Nested Catalog Support
Updated endpoints and types to allow catalogs to be nested within other catalogs. This includes adding a new `catalog_path` variable to most endpoints to allow more specific identification of catalogs for nesting functionality.
Also updating link functionality to ensure all nested items can be retrieved as expected via the correct URL.

### CODE REVIEW

File: stac_fastapi/api/tests/benchmarks.py [hunk 1]
```diff
@@ -121,15 +123,27 @@ def all_catalogs(self, **kwargs) -> stac_types.Catalogs:
             ],
         )
 
+    # def all_nested_catalogs(self, **kwargs) -> stac_types.Catalogs:
```
Comment (sgillies-tpzuk) [inline stac_fastapi/api/tests/benchmarks.py]: Can this commented out code be removed?

File: stac_fastapi/types/stac_fastapi/types/core.py [hunk 2]
```diff
@@ -590,9 +608,20 @@ def all_catalogs(self, **kwargs) -> stac_types.Catalogs:
         """
         ...
 
+    # @abc.abstractmethod
```
Comment (sgillies-tpzuk) [inline stac_fastapi/types/stac_fastapi/types/core.py]: Can this commented out code be removed?

File: stac_fastapi/types/stac_fastapi/types/core.py [hunk 3]
```diff
@@ -897,6 +928,17 @@ async def all_catalogs(self, **kwargs) -> stac_types.Catalogs:
...
```
Comment (sgillies-tpzuk) [inline stac_fastapi/types/stac_fastapi/types/core.py]: Can this commented out code be removed?

**Review by sgillies-tpzuk — APPROVED** [review]

File: stac_fastapi/api/tests/benchmarks.py (hunk 1 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/api/tests/benchmarks.py]: removed

File: stac_fastapi/types/stac_fastapi/types/core.py (hunk 2 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/types/stac_fastapi/types/core.py]: removed

File: stac_fastapi/types/stac_fastapi/types/core.py (hunk 3 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/types/stac_fastapi/types/core.py]: removed

**Pull request merged**